### ISSUE TITLE
Problem with the deployment in Digital Ocean

### ISSUE THREAD

**Issue by Mullycap** [mirumee/saleor#391]
Hi:
We are a group of young developers that have used your project of saleor. In our development environment we didn’t have any problem with our customization of the shop but when we deployed to digital ocean we started having problems with the session cart.
When we add a product the redirect works properly but sometimes the item is added to the cart and sometimes no, apart from that another unusual behaviour is that if we refresh sometimes the cart is correct and other times shows the incorrect amount, and finally in the cart index url we can not erase any of the items properly.
Our installation has the following:
Babel==1.3
BabelDjango==0.2.2
Cheetah==2.4.4
Django==1.8.5
Landscape-Client==14.12
Markdown==2.6.4
PAM==0.4.2
Pillow==3.0.0
PyJWT==1.4.0
PyOpenGL==3.0.2
PyYAML==3.10
Pyrex==0.9.8.5
South==0.7.5
Twisted-Core==13.2.0
Unidecode==0.04.18
apt-xapian-index==0.45
argparse==1.2.1
boto==2.38.0
braintree==3.21.0
chardet==2.0.1
cloud-init==0.7.5
colorama==0.2.5
configobj==4.7.2
cssselect==0.9.1
cssutils==1.0.1
dj-database-url==0.3.0
django-absolute==0.3
django-emailit==0.2.2
django-materializecss-form==0.1
django-model-utils==2.0.3
django-mptt==0.7.4
django-offsite-storage==0.0.6
django-payments==0.8.1
django-prices==0.4.1
django-selectable==0.8.0
django-storages==1.1.8
django-versatileimagefield==1.0.6
django-wysiwyg-redactor==0.4.9
djangorestframework==3.1.3
e==1.4.5
fake-factory==0.5.3
gevent==1.0
google-measurement-protocol==0.1.4
greenlet==0.4.2
gunicorn==17.5
gyp==0.1
html5lib==0.999
jsonfield==1.0.3
jsonpatch==1.3
jsonpointer==1.0
lxml==3.5.0
oauth==1.0.1
premailer==2.9.6
prettytable==0.7.2
prices==0.5.4
psycopg2==2.4.5
py==1.4.20
pyOpenSSL==0.13
pycryptodome==3.3.1
pyserial==2.6
pytest==2.5.1
python-apt==0.9.3.5ubuntu1
python-debian==0.1.21-nmu2ubuntu2
pytz==2015.7
requests==2.2.1
saleor==0.1.0a0
satchless==1.1.3
sendgrid==1.5.14
six==1.5.2
smtpapi==0.2.0
ssh-import-id==3.21
stripe==1.27.1
suds-jurko==0.6
urllib3==1.7.1
virtualenv==1.11.4
wheel==0.24.0
whitenoise==2.0.4
wsgiref==0.1.2
xmltodict==0.9.2
zope.interface==4.0.5

**Comment by patrys** [comment]
Are you using cache for session storage? Are there multiple Django processes or multiple droplets? If so, do you use a shared cache server between them? Django defaults to caching in local process memory which mean each process would have its own idea of what a particular session contains. This is not a problem with `./mange.py runserver` as it's always using a single process.

**Comment by patrys** [comment]
If you want to give back, we'd love to hear the story behind your store. How you built it, what works and what doesn't, what are the areas where we could improve? Cheers!

**Comment by Mullycap** [comment]
Hi: 
We are developing the store for a friend that sells Gourmet products online (from Spain). We developed it  using you're project like base, then we develop a customized frontend templates using Yoman creating a webapp with grunt and bootstrap. 
With this we develop extra modules needed for showing the events that the company organize and recipes with their products, apart form that we develop static section (contact, home...)
Then we modify some things from you're project to fullfil the requirements, for example if a distributor logs in the web we show the stock price, and calculate the total price using the stock price. In the delivery we added a method to calculate the shipment price depending on the weight of the package and the distance between the shop and the delivery destination. Then we made some changes to put a price from which the shipment is free, we added a parameter in the settings that determine that price. 
Now that the the server is running we are trying to make work the payments in the server and thinking how to add the possibility to add coupons in the checkout to make discounts in the final price.
For server we used digital ocean because we where used to use it, to make it work we had to modify the gunicorn to use your wsgi because was a custom droplet with a basic django project.
I will start with the things that didn't work that is only one, the fixed product discount launch a error.
Then the things that works, the dashboard is amazing, easy to use and very fluid and quick. In the more technical part, the code is very clear to add things and learn a little bit, we just started with python and Django.
For improve we don't think about anything now, only the things we needed to add so we needed to do anything, but we think that could be a good idea where a price and the type of packaging for them, like quantity of items for box for example.
I think that I didn't forgot anything now we have to ensure that the payments work properly in the server now that we have the cart and checkout working properly.
Again thanks for making this project opensource
